panther_description/urdf/body.urdf.xacro

```diff
@@ -43,51 +50,51 @@
     <joint name="body_to_imu_joint" type="fixed">
       <origin xyz="${imu_pos_x} ${imu_pos_y} ${imu_pos_z}"
         rpy="${imu_rot_r} ${imu_rot_p} ${imu_rot_y}" />
-      <parent link="body_link" />
-      <child link="imu_link" />
+      <parent link="${ns}body_link" />
+      <child link="${ns}imu_link" />
     </joint>
 
-    <link name="imu_link" />
+    <link name="${ns}imu_link" />
 
     <joint name="body_to_front_bumper_joint" type="fixed">
       <origin xyz="0.362 0.0 0.0" rpy="0.0 0.0 0.0" />
-      <parent link="body_link" />
-      <child link="front_bumper_link" />
+      <parent link="${ns}body_link" />
+      <child link="${ns}front_bumper_link" />
     </joint>
 
-    <link name="front_bumper_link" />
+    <link name="${ns}front_bumper_link" />
 
     <joint name="front_bumper_to_front_light_joint" type="fixed">
       <origin xyz="0.0 0.0 0.0185" rpy="${pi/2} 0.0 ${pi/2}" />
-      <parent link="front_bumper_link" />
-      <child link="front_light_link" />
+      <parent link="${ns}front_bumper_link" />
+      <child link="${ns}front_light_link" />
     </joint>
 
-    <link name="front_light_link" />
+    <link name="${ns}front_light_link" />
 
     <joint name="body_to_rear_bumper_joint" type="fixed">
       <origin xyz="-0.362 0.0 0.0" rpy="0.0 0.0 ${pi}" />
-      <parent link="body_link" />
-      <child link="rear_bumper_link" />
+      <parent link="${ns}body_link" />
+      <child link="${ns}rear_bumper_link" />
     </joint>
 
-    <link name="rear_bumper_link" />
+    <link name="${ns}rear_bumper_link" />
 
     <joint name="rear_bumper_to_rear_light_joint" type="fixed">
       <origin xyz="0.0 0.0 0.0185" rpy="${pi/2} 0.0 ${pi/2}" />
-      <parent link="rear_bumper_link" />
-      <child link="rear_light_link" />
+      <parent link="${ns}rear_bumper_link" />
+      <child link="${ns}rear_light_link" />
     </joint>
 
-    <link name="rear_light_link" />
+    <link name="${ns}rear_light_link" />
 
     <joint name="body_to_cover_joint" type="fixed">
```

Comment:
```suggestion
    <joint name="${ns}body_to_${ns}cover_joint" type="fixed">
```

Reply:
I understand this suggestions, but I notice that there is no conflict when you leave it like it is, which is much cleaner.

There is also secound option, we can add tf_prefix param into node robot_description. There is only one small disadvantage, because you need also to add tf_prefix in RVIZ RobotModel.